designer_v2/lib/features/dashboard/dashboard_state.dart

```diff
@@ -0,0 +1,52 @@
+import 'package:equatable/equatable.dart';
+import 'package:studyu_core/core.dart';
+import 'package:studyu_designer_v2/features/dashboard/studies_filter.dart';
+import 'package:supabase_flutter/supabase_flutter.dart';
+
+enum DashboardStatus { initial, loading, success, failure }
+
+class DashboardState extends Equatable {
+  static const defaultFilter = StudiesFilter.owned;
+
+  const DashboardState({
+    this.status = DashboardStatus.initial,
+    this.studies = const [],
+    this.studiesFilter = defaultFilter,
+    required this.currentUser,
+  });
+
+  final DashboardStatus status;
+
+  /// The list of studies that can be accessed by the current user
+  final List<Study> studies;
+
+  /// Currently selected filter to be applied to the list of studies
+  /// in order to determine the [visibleStudies]
+  final StudiesFilter studiesFilter;
+
+  /// Currently authenticated user (used for filtering studies)
+  final User currentUser;
+
+  /// The currently visible list of studies as by the selected filter
+  List<Study> get visibleStudies =>
+      studiesFilter.apply(studies: studies, user: currentUser).toList();
```

Comment:
Shouldn't `visibleStudies` be equal to `all` studies?

Reply:
Only if `studyFilter == StudyFilter.all` because the filter is a no-op in that case (see StudyFilter, the logic here is copied from the old designer). This is in fact how the public Study Registry page is implemented currently (not ideal for obvious reasons).

Normally `visibleStudies` is a subset of all studies. The full set of studies that the user can access & that is returned from querying the Studies table in Supabase is stored in `DashboardState.studies` (fetched & synced from the repository stream). The `studiesFilter` is controlled by the current dashboard route (e.g. navigating to "Shared Studies" will set the `studiesFilter` to `StudyFilter.shared`). Because `studiesFilter` is used in the state's equality check (see `Equatable.props`), updating the state with the new `studiesFilter` will result in a rerender of the widgets that are watching the `DashboardController` (this is just how StateNotifier works in Riverpod to enable reactivity). When the widget is rerendered, it calls the `visibleStudies` getter (dynamically applying the new filter to the same set of all studies that are accessible to the user). This way we can reuse the same widget to display a different subset of studies on different routes (/studies/owned, /studies/shared, /studies/registry).